test/integration/apiserver/apply/reset_fields_test.go

```diff
@@ -298,3 +297,31 @@ func TestApplyResetFields(t *testing.T) {
 		}
 	}
 }
+
+func expectConflict(t *testing.T, objRet *unstructured.Unstructured, err error, dynamicClient dynamic.Interface, resource schema.GroupVersionResource, namespace, name, where string) {
+	if err != nil && strings.Contains(err.Error(), "conflict") {
+		return
+	}
+	which := "returned"
+	var gotten string
+	var err2 error
+	if objRet == nil {
```

Comment:
You're never checking `if check != nil` to see if there is a non-conflict error, but this way of checking for errors is not idiomatic.

Reply:
In this situation, where clearly something unexpected is happening, I think it best to not make more assumptions (such as `objRet == nil` if and only iff `err != nil`) than necessary.  The code here is fully revealing about both `err` and `objRet` without making unnecesary assumptions.